streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -721,6 +789,28 @@ private static Number getMax(final String key) {
         return Integer.parseInt(key.split("-")[1]);
     }
 
+    private static VerificationResult performEosVerification(final boolean eosEnabled, final String kafka) {
+        if (!eosEnabled) {
+            return new VerificationResult(true, "EOS verification skipped");
+        }
+        
+        final Properties eosProps = new Properties();
+        eosProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "transaction-verifier");
+        eosProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka);
+        eosProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+        eosProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+        eosProps.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.toString());
+        
+        try (final KafkaConsumer<byte[], byte[]> eosConsumer = new KafkaConsumer<>(eosProps)) {
+            verifyAllTransactionFinished(eosConsumer, kafka);
+            return new VerificationResult(true, "EOS verification passed");
```

Comment:
If we are working with `VerificationResult`, would it be better to let `verifyAllTransactionFinished` return what we need? Or at least a boolean, to avoid unnecessary exception handling?